networkx/algorithms/community/demon.py

```diff
@@ -0,0 +1,296 @@
+# -*- coding: utf-8 -*-
+#    Copyright (C) 2015
+#    All rights reserved.
+#    BSD license.
+
+"""
+Flat Merge version of DEMON (Democratic Estimate of the Modular Organization of
+a Network) algorithm as described in:
+
+Michele Coscia, Giulio Rossetti, Fosca Giannotti, Dino Pedreschi:
+DEMON: a local-first discovery method for overlapping communities.
+KDD 2012:615-623
+
+DEMON uses a local-first approach to community discovery were each node
+democratically votes for the communities that it sees in its surroundings. It
+provides a deterministic, fully incremental method for community discovery that
+has a limited time complexity.
+
+Based on the implementation by Giulio Rossetti <giulio.rossetti@isti.cnr.it>:
+http://www.michelecoscia.com/wp-content/uploads/2013/07/demon_py.zip
+"""
+
+from __future__ import division
+import networkx as nx
+import random
+
+__all__ = ["demon_communities"]
+
+
+def demon_communities(G, epsilon=0.25, min_community_size=3, weight=None):
+    """Finds the communities in a graph using the label Democratic Estimate of
+    the Modular Organization of a Network (DEMON) algorithm[1]_. This method
+    uses a local-first approach to community discovery were each node
+    democratically votes for the communities that it sees in its surroundings.
+    Communities in this algorithm can be overlapping
+
+    Algorithm:
```

Comment:
Don't include this in the documentation string, it is not important for the user. The asymptotic running time may, however, be important to the user.